docs/grammar.txt

```diff
@@ -58,7 +58,7 @@ ArrayTypeName = TypeName '[' Expression? ']'
 FunctionTypeName = 'function' FunctionTypeParameterList ( 'internal' | 'external' | StateMutability )*
                    ( 'returns' FunctionTypeParameterList )?
 StorageLocation = 'memory' | 'storage'
-StateMutability = 'pure' | 'constant' | 'view' | 'payable'
+StateMutability = 'pure' | 'view' | 'payable'
```

Comment:
This cannot be removed here.

Reply:
OK - the idea of this (and the one's below) was to already update the documentation *as if constant had already been removed* and only mentioning it's existence in a note, but you're probably right that it's better to postpone that until the breaking change.